plugins/flytekit-comet-ml/flytekitplugins/comet_ml/tracking.py

```diff
@@ -0,0 +1,146 @@
+    hostname = hostname.replace("-", "")
+    suffix = _generate_suffix_with_length_10(project_name, workspace)
+    return f"{hostname}{suffix}"
+
+
+class comet_ml_init(ClassDecorator):
+    COMET_ML_PROJECT_NAME_KEY = "project_name"
+    COMET_ML_WORKSPACE_KEY = "workspace"
+    COMET_ML_EXPERIMENT_KEY_KEY = "experiment_key"
+    COMET_ML_URL_SUFFIX_KEY = "link_suffix"
+    COMET_ML_HOST_KEY = "host"
+
+    def __init__(
+        self,
+        task_function: Optional[Callable] = None,
+        project_name: Optional[str] = None,
+        workspace: Optional[str] = None,
+        experiment_key: Optional[str] = None,
+        secret: Optional[Union[Secret, Callable]] = None,
+        host: str = "https://www.comet.com",
+        **init_kwargs: dict,
+    ):
+        """Comet plugin.
+        Args:
+            task_function (function, optional): The user function to be decorated. Defaults to None.
+            project_name (str): Send your experiment to a specific project. (Required)
+            workspace (str): Attach an experiment to a project that belongs to this workspace. (Required)
+            experiment_key (str): Experiment key.
+            secret (Secret or Callable): Secret with your `COMET_API_KEY` or a callable that returns the API key.
+                The callable takes no arguments and returns a string. (Required)
+            host (str): URL to your Comet service. Defaults to "https://www.comet.com"
+            **init_kwargs (dict): The rest of the arguments are passed directly to `comet_ml.init`.
+        """
+        if project_name is None:
+            raise ValueError("project_name must be set")
+        if workspace is None:
+            raise ValueError("workspace must be set")
+        if secret is None:
+            raise ValueError("secret must be set")
+
+        self.project_name = project_name
+        self.workspace = workspace
+        self.experiment_key = experiment_key
+        self.secret = secret
+        self.host = host
+        self.init_kwargs = init_kwargs
+
+        super().__init__(
+            task_function,
+            project_name=project_name,
+            workspace=workspace,
+            experiment_key=experiment_key,
+            secret=secret,
+            host=host,
+            **init_kwargs,
+        )
+
+    def execute(self, *args, **kwargs):
+        ctx = FlyteContextManager.current_context()
+        is_local_execution = ctx.execution_state.is_local_execution()
+
+        default_kwargs = self.init_kwargs
+        init_kwargs = {
+            "project_name": self.project_name,
+            "workspace": self.workspace,
+            **default_kwargs,
+        }
+
+        if is_local_execution:
+            # For local execution, always use the experiment_key. If `self.experiment_key` is `None`, comet_ml
+            # will generate it's own key
+            if self.experiment_key is not None:
+                init_kwargs["experiment_key"] = self.experiment_key
```

Comment:
is api key not required for local executions?

Reply:
It's required, I use `export COMET_API_KEY="xxxx"` to test it in local execution.